src/v/cluster/scheduling/leader_balancer.h

```diff
@@ -62,6 +62,13 @@ class leader_balancer {
      */
     static constexpr clock_type::duration leader_transfer_rpc_timeout = 30s;
 
+    /*
+     * limits the total in flight leadership transfers (those where the metadata
+     * has yet to propagate across the cluster) to this factor per shard in the
+     * cluster.
+     */
+    static constexpr size_t transfer_limit_per_shard = 10;
```

Comment:
should probably be a config option.

Reply:
I think we can either set this to something quite high (e.g. 512) so that it's only hit in seriously busy cases, or go ahead and make it configurable.  Probably not a lot of effort to make it a config (make sure the visibility level is `tunable` + use a bounded_property with some reasonable bounds)